kubectl-pg/cmd/scale.go

```
@@ -0,0 +1,144 @@
+		confirmAction(clusterName, namespace)
+	}
+	postgresql.Kind = "postgresql"
+	postgresql.APIVersion = "acid.zalan.do/v1"
+	UpdatedPostgres, err := postgresConfig.Postgresqls(namespace).Update(postgresql)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if UpdatedPostgres.ResourceVersion != postgresql.ResourceVersion {
+		fmt.Printf("scaled postgresql %s/%s to %d instances\n", UpdatedPostgres.Namespace, UpdatedPostgres.Name, UpdatedPostgres.Spec.NumberOfInstances)
+		return
+	}
+	fmt.Printf("postgresql %s is unchanged.\n", postgresql.Name)
+}
+
+func allowedMinMaxInstances(config *rest.Config) (int32, int32){
+	k8sClient,err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	operator,err :=k8sClient.AppsV1().Deployments(getCurrentNamespace()).Get("postgres-operator",metav1.GetOptions{})
+	if err!= nil {
+		log.Fatal(err)
+	}
+	operatorContainer:=operator.Spec.Template.Spec.Containers
+	var configMapName, operatorConfigName string
+	minInstances := -1
+	maxInstances := -1
+	for _,envData := range operatorContainer[0].Env {
+		if envData.Name == "CONFIG_MAP_NAME"  {
+			configMapName = envData.Value
+		}
+		if envData.Name == "POSTGRES_OPERATOR_CONFIGURATION_OBJECT" {
+			operatorConfigName = envData.Value
+		}
+	}
+	if operatorConfigName == "" {
+		configMap, err := k8sClient.CoreV1().ConfigMaps(getCurrentNamespace()).Get(configMapName, metav1.GetOptions{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		configMapData := configMap.Data
+		for key, value := range configMapData {
+			if key == "min_instances" {
+				minInstances,_ = strconv.Atoi(value)
+			}
+			if key == "max_instances" {
+				maxInstances,_ = strconv.Atoi(value)
+			}
+		}
+	} else if configMapName == "" {
+		pgClient,err :=	PostgresqlLister.NewForConfig(config)
+		if err != nil {
+			log.Fatal(err)
+		}
+		operatorConfig,_ := pgClient.OperatorConfigurations("s").Get(operatorConfigName,metav1.GetOptions{})
```

do not ignore errors with `_`

---

acknowledged